src/main/java/com/redhat/devtools/lsp4ij/settings/ui/AbstractServerMappingTablePanel.java

```diff
@@ -61,25 +62,26 @@ public List<ServerMappingSettings> getServerMappings() {
     }
 
     private AnActionButtonRunnable addData() {
-        return new AnActionButtonRunnable() {
-            @Override
-            public void run(AnActionButton anActionButton) {
-                ServerMappingSettings newMapping = createServerMappingSettings();
-                table.getListTableModel().addRow(newMapping);
-                fireStateChanged();
-            }
+        return anActionButton -> {
+            ServerMappingSettings newMapping = createServerMappingSettings();
+            table.getListTableModel().addRow(newMapping);
+            fireStateChanged();
+            // Immediately select the first cell in the new row
+            ApplicationManager.getApplication().invokeLater(() -> {
+                int newRowIndex = table.getRowCount() - 1;
+                table.getSelectionModel().setSelectionInterval(newRowIndex, newRowIndex);

```

Comment:
As mentioned in the PR itself, this could instead just start editing the first cell in the new row. I tried that and felt like it was a little too aggressive, but I'm open to other opinions, of course.